File: Project Code/Calculator/MiniCalculator.java
import java.util.Arrays;

public class MiniCalculator {

	public static final String[] ZERO = {"zero", "oh"};
	public static final String[] ONES = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
	public static final String[] TEENS = {"ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
	public static final String[] TENS = {"twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};
	public static final String[] MAGNITUDES = {"hundred", "thousand", "million", "billion"};

	public static final String[][] ALL = {ZERO, ONES, TEENS, TENS, MAGNITUDES};

	private String input;
	private double[] vals;
	private String[] ops;

	public boolean decode() { // if this returns false, input is not a mathematical equation. if true, then this function turns String input into two arrays: values and operations
		// start by removing all "and"'s from the input since they are redundant, as well as "by" (in the case of "divided by")
		input.replaceAll("and ", "");
		String[] inputSplit = input.split(" ");
		vals = new double[(inputSplit.length + 1) / 2];
	    int	valCount = 0;
		double temp = 0; // stores cumulative value until the next operation is found (for switch statement)
		double tempMag = 0; // stores the most recent value that is to be either gaining magnitude or being added
		for (int i = 0; i < inputSplit.length; i++) { // cycling through all inputted words
			for (int j = 0; j < ALL.length; j++) { // cycling through all arrays (but not through the contents of each array)
				if (Arrays.asList(ALL[j]).indexOf(inputSplit[i]) != -1) { // indexing the content of each array for the current word
					int col = Arrays.asList(ALL[j]).indexOf(inputSplit[i]);
					switch (j) {
						case 0: tempMag =  0; // j = 0 is array ZERO (if we are in the context of one oh one (which is the MUCH MORE LIKELY USE FOR 0), this should be multiplying by ten instead)
								break;
						case 1: tempMag = col + 1; // j = 1 is array ONES, where the array index # is always 1 less than the value represented
								break;
						case 2: tempMag = 10 + col; // j = 2 is array TEENS, where the value represented is always 10 + the array index #
								break;
						case 3: tempMag = (col + 2) * 10; // j = 3 is array TENS
								break;
						case 4: if (col == 0) {
									tempMag *= 100;
								} else {
									tempMag *= Math.pow(1000, col); // col corresponds to mag i.e. col = 1: 1000, col = 2: 1000 * 1000 = 1000000 
								}
								break; // no need to increment valCount. magnitude only affects the most recent value and does not add any new ones
					}
					if (i + 1 == inputSplit.length) { // meaning next is operation, so copy temp into th vals array. or, if there is no next value i.e. i is on the last iteration, copy.
						temp += tempMag; // if we do not have this, the very last value of inputSplit will be missed. it will be stored in tempMag because of the switch statement, but it is not stored into temp
						vals[valCount] = temp;
						valCount++;
						temp = 0;
						tempMag = 0;
					} else { // if next input is magnitude, then this shouldn't be allowed to add it as tempMag needs to individually be magnified. if not, thten just add them together.
						if (j != 5 && Arrays.asList(MAGNITUDES).indexOf(inputSplit[i + 1]) == -1) {
							temp += tempMag; // temp stores the cumulative value. note that, if the next input value is a magnitude, this does not run, so tempMag is multiplied by a certain magnitude and then added
						}
					}
					//for (int k = 0; k < ops.length; k++) { 
					//	System.out.println(vals[k] + "next is" + vals[k+1] + " hello " + ops[k]); 
					//}
					break; // start searching for the next word, since this one was already found. also prevents program from returning false even if a word was found
				}
			}
			//return false;// leaving this false gives the program some leeway.. not sure whats the best way to apparoach this //BUGGED HERE, THIS WILL MAKE THE PROGRAM ALWAYS RETURN FALSE FOR SOME REASON // if the for loop is not broken, that must mean no word was matched. so this is not a mathermatical statement.
		}
		return true;
	}
	
	public double getValue() {
		return vals[0]; // values all shifted down to the first index.
	}

	public MiniCalculator(String input) {
		this.input = input;
	}

	public static void main(String[] args) { // if keyword == what is... trouble doing composite brackets
		Calculator calc = new Calculator("three hundred million four hundred thousand nine hundred and forty two");
		if (calc.decode() == true) {
			System.out.println(calc.getValue());
		} else {
			System.out.println("Not a number!"); //note this never triggers because no return false statement (yet) // for the actual thing, this defaults to google search instead
		}
	}	
}

File: Project Code/Main/MiniCalculator.py
import string
from decimal import Decimal

class MiniCalculator:
	ZERO = ["zero", "oh"]
	ONES = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
	TEENS = ["ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"]
	TENS = ["twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"]
	MAGNITUDES = ["hundred", "thousand", "million", "billion"]

	inputPhrase = None

	def __init__(self, inputPhrase=None):
		if inputPhrase!=None:
			self.inputPhrase = inputPhrase

	def newInput(self, inputPhrase):
		self.inputPhrase = inputPhrase

	def decode(self):
		self.inputPhrase = self.inputPhrase.replace(" and ", " ")
		inputPhraseSplit = self.inputPhrase.split(" ")
		cumulativeValue = 0 
		tempMag = 0
		insertMode = False

		checkLength = len(inputPhraseSplit)-1
		if "point" in inputPhraseSplit:
			checkLength = inputPhraseSplit.index("point")
		if not set(inputPhraseSplit[:checkLength]).isdisjoint(self.ZERO):
			insertMode = True
		else:
			for i in range(checkLength):
				if inputPhraseSplit[i] in self.ONES and inputPhraseSplit[i+1] in self.ONES:
					insertMode = True
					break

		for i in range(len(inputPhraseSplit)):
			if inputPhraseSplit[i] == "point": #everthing after the point should be said as single digits
				decimalPlace = 1
				for j in range(i+1, len(inputPhraseSplit)):
					if inputPhraseSplit[j] in self.ONES:
						cumulativeValue+=Decimal(self.ONES.index(inputPhraseSplit[j])+1)/Decimal(pow(10,decimalPlace)) #solves floating point errors
					decimalPlace+=1
				break
			else:
				if insertMode == True: #IF THEY HAVE A ZERO ANYWHERE or if two "ones" are beside one another anywhere, THEY LIEKLY MEAN TO SAY DIGITS ONE BY ONE 
					cumulativeValue*=10
					if inputPhraseSplit[i] in self.ONES: 
						cumulativeValue+=self.ONES.index(inputPhraseSplit[i])+1
				else:
					if inputPhraseSplit[i] in self.ONES:
						col = self.ONES.index(inputPhraseSplit[i])
						tempMag += col + 1 # j = 1 is array self.ONES, where the array index # is always 1 less than the value represented
					elif inputPhraseSplit[i] in self.TEENS:
						col = self.TEENS.index(inputPhraseSplit[i])
						tempMag += 10 + col # j = 2 is array self.TEENS, where the value represented is always 10 + the array index #
					elif inputPhraseSplit[i] in self.TENS:
						col = self.TENS.index(inputPhraseSplit[i])
						tempMag += (col + 2) * 10 # j = 3 is array self.TENS
					elif inputPhraseSplit[i] in self.MAGNITUDES:
						col = self.MAGNITUDES.index(inputPhraseSplit[i])
						if (col == 0):
							tempMag *= 100
						else:
							tempMag *= pow(1000, col)
					if i+1 == len(inputPhraseSplit): 
						cumulativeValue += tempMag
					elif inputPhraseSplit[i+1] not in self.MAGNITUDES: #case where we have a number with a magnitude after i.e. two hundred. we don't want to add it to the cumulative total just yet
						if inputPhraseSplit[i] in self.TENS and i+2 < len(inputPhraseSplit): #case where we have a joint number with a magnitude after i.e. twenty two thousand
							if inputPhraseSplit[i+2] not in self.MAGNITUDES:
								cumulativeValue += tempMag
								tempMag = 0
						elif inputPhraseSplit[i] == "hundred":  #there are two possible ways to use hundred: one is to get the "222" part of "22200" and another would be to get the "200" part of "1200"
							if i+2 < len(inputPhraseSplit):
								if inputPhraseSplit[i+2] not in self.MAGNITUDES:
									if i+3 < len(inputPhraseSplit):
									 	if inputPhraseSplit[i+3] not in self.MAGNITUDES:
												cumulativeValue += tempMag
												tempMag = 0
									else:
										cumulativeValue += tempMag 
										tempMag = 0	
							else: 
								cumulativeValue += tempMag 
								tempMag = 0				
						else: 
							cumulativeValue += tempMag 
							tempMag = 0
		return cumulativeValue